src/ray/common/ray_config_def.h

```diff
@@ -454,6 +454,16 @@ RAY_CONFIG(uint64_t, gcs_service_address_check_interval_milliseconds, 1000)
 /// The batch size for metrics export.
 RAY_CONFIG(int64_t, metrics_report_batch_size, 100)
 
+/// The interval duration for which task state events will be reported to GCS.
+/// The reported data should only be used for observability.
+/// Setting the value to 0 to disable the task event recording and reported.
+RAY_CONFIG(int64_t, task_events_report_interval_ms, 0)
+
+/// The number of tasks tracked in GCS for task state events. Events from additional
+/// tasks will be dropped.
+/// Set to -1 for allowing unlimited events stored in GCS.
+RAY_CONFIG(int64_t, task_events_max_num_task_in_gcs, 100000);
```

Comment:
Each task's events takes less than 1KB (avg 200bytes -> 400bytes), so I guess we could have 100K stored => 100MiB max.

Reply:
This sounds good for now. We can reconsider in the future with per-job caps.